test_rclcpp/test/test_allocator.cpp

```diff
@@ -385,59 +383,52 @@ TEST(CLASSNAME(test_allocator, RMW_IMPLEMENTATION), unique_ptr) {
-      alloc);
-  auto subscriber = node->create_subscription<test_rclcpp::msg::UInt32>(
-    "test_allocator_unique_ptr", 10, callback, nullptr, false, msg_mem_strat, alloc);
-  std::shared_ptr<rclcpp::memory_strategy::MemoryStrategy> memory_strategy =
-    std::make_shared<AllocatorMemoryStrategy<UInt32Allocator>>(alloc);
-
-  rclcpp::executors::SingleThreadedExecutor executor(memory_strategy);
-  executor.add_node(node);
+  auto subscriber = node_->create_subscription<test_rclcpp::msg::UInt32>(
+    "allocator_unique_ptr", 10, callback, nullptr, false, msg_memory_strategy_, alloc);
 
-  size_t test_initialization_allocs = num_allocs;
-  size_t test_initialization_deallocs = num_deallocs;
+  InstrumentedAllocator<test_rclcpp::msg::UInt32> msg_alloc;
 
   rclcpp::utilities::sleep_for(std::chrono::milliseconds(1));
   // After test_initialization, global new should only be called from within InstrumentedAllocator.
   test_init = true;
   for (uint32_t i = 0; i < iterations; i++) {
     auto msg = std::unique_ptr<test_rclcpp::msg::UInt32, UInt32Deleter>(
-      std::allocator_traits<UInt32Allocator>::allocate(*alloc.get(), 1));
+      std::allocator_traits<UInt32Allocator>::allocate(msg_alloc, 1));
     msg->data = i;
-    publisher->publish(msg);
+    publisher_->publish(msg);
     rclcpp::utilities::sleep_for(std::chrono::milliseconds(1));
-    executor.spin_some();
+    executor_->spin_some();
   }
   test_init = false;
+  EXPECT_FALSE(fail);
+  fail = false;
+}
 
-  size_t runtime_allocs = num_allocs - test_initialization_allocs;
-  size_t runtime_deallocs = num_deallocs - test_initialization_deallocs;
-
-  EXPECT_EQ(global_runtime_allocs, 0);
-  EXPECT_EQ(global_runtime_deallocs, 0);
-
-  printf("Global new was called outside of execution: %zu\n", global_allocs);
-  printf("Global delete was called outside of execution: %zu\n",
-    global_deallocs);
-  printf("Allocator new was called during initalization: %zu\n", test_initialization_allocs);
-  printf("Allocator delete was called during initalization: %zu\n", test_initialization_deallocs);
-
-  printf("Calls to global new after spinning some: %zu\n", global_runtime_allocs);
-  printf("Calls to global delete after spinning some: %zu\n", global_runtime_deallocs);
+void print_help()
+{
+  printf("--all-tokens: Do not ignore middleware tokens.\n");
+  printf("--verbose: Report stack traces and allocation statistics.\n");
+}
 
-  printf("Calls to allocator new after spinning some: %zu\n", runtime_allocs);
-  printf("Calls to allocator delete after spinning some: %zu\n", runtime_deallocs);
+int main(int argc, char ** argv)
+{
+  rclcpp::init(argc, argv);
+  ::testing::InitGoogleTest(&argc, argv);
+  // argc and argv are modified by InitGoogleTest
+  std::vector<std::string> args(argv + 1, argv + argc);
+  bool help = (std::find(args.begin(), args.end(), "--help") != args.end());
+  if (help || argc == 0) {
```

Comment:
Since help is also not used anywhere else this could be boiled down to:

    if (std::find(args.begin(), args.end(), "--help") != args.end()) {
      print_help();